### PR TITLE
EVG-5695 Refactor next task based on flags

### PR DESCRIPTION
**Pull request by bsamek**

### CODE REVIEW

File: service/api_task.go [hunk 1]
```diff
@@ -317,7 +320,36 @@ func assignNextAvailableTask(taskQueue *model.TaskQueue, currentHost *host.Host)
...
+						"host":    currentHost.Id,
+						"message": "problem refreshing taskQueueService",
+						"spec":    spec,
+					}))
+					return nil, errors.Wrap(err, "problem refreshing taskQueueService")
+				}
+			}
+			queueItem, err = taskQueueService.FindNextTask(currentHost.Distro.Id, spec)
+			if err != nil {
+				grip.Alert(message.WrapError(err, message.Fields{
```
Comment (tychoish) [inline service/api_task.go]: grip.Critical?

File: service/api_task.go [hunk 2]
```diff
@@ -317,7 +320,36 @@ func assignNextAvailableTask(taskQueue *model.TaskQueue, currentHost *host.Host)
 	// continue must be preceded by dequeueing the current task from the
 	// queue to prevent an infinite loop.
 	for taskQueue.Length() != 0 {
-		queueItem := taskQueue.FindNextTask(spec)
+		var queueItem *model.TaskQueueItem
+		var err error
+		switch currentHost.Distro.PlannerSettings.Version {
+		case evergreen.PlannerVersionTunable:
+			if taskQueueService == nil {
```
Comment (tychoish) [inline service/api_task.go]: can we make sure this is never true by putting it in the initialization of the ui server?

File: service/api_task.go [hunk 3]
```diff
@@ -317,7 +320,36 @@ func assignNextAvailableTask(taskQueue *model.TaskQueue, currentHost *host.Host)
...
+					grip.Alert(message.WrapError(err, message.Fields{
+						"distro":  currentHost.Distro.Id,
+						"host":    currentHost.Id,
+						"message": "problem refreshing taskQueueService",
+						"spec":    spec,
+					}))
+					return nil, errors.Wrap(err, "problem refreshing taskQueueService")
+				}
+			}
+			queueItem, err = taskQueueService.FindNextTask(currentHost.Distro.Id, spec)
```
Comment (tychoish) [inline service/api_task.go]: Maybe write a wrapper (or a method somewhere) that does the "refresh if needed?" logic or just be able to call refresh and have it noop always?

File: service/api_task.go (hunk 2 above)
Comment (bsamek) [reply service/api_task.go]: done

File: service/api_task.go (hunk 3 above)
Comment (bsamek) [reply service/api_task.go]: done

File: service/api_task.go (hunk 1 above)
Comment (bsamek) [reply service/api_task.go]: done

**Pull request merged**